### ISSUE TITLE
Yezidi characters cannot be ligated by ccmp

### ISSUE THREAD

**Issue by cheonhyeongsim** [harfbuzz/harfbuzz#4667]
[hebr.zip](https://github.com/harfbuzz/harfbuzz/files/15016932/hebr.zip)

Harfbuzz could not ligate the cross-cluster sequences of the Yezidi script. In the original proposal of Yezidi, it mentioned some historical ligatures. Indeed, using `hlig` would be better for the historical ligatures, but I also want to use them in plain texts, so I use `ccmp` to ligate the ZWJ-sequence (in order not to ligate the default case). For example, `10EA0 200D 10E86` should be ligated. If I use `rclt`, the ligature would work well, but if I use `ccmp`, nothing happens. The font file is uploaded in the attached zip file.

I believe this is no longer an issue related to Chrome but related to Harfbuzz - I tested it both in Crowbar and in hb-shape.

Yezidi is not an Indic script, so why `ccmp` could not ligate the cross-cluster sequences? The Hebrew script is very similar to the Yezidi script that, they are both RTL non-cursive scripts. The attached font also contains the Hebrew script, and you could see that `05D0 200D 05DC` works well - that is also a ZWJ-sequence under `ccmp`.

By the way, I would like to say that the Khitan Small Script has the same issue. I even thought that Harfbuzz mistakenly treated Khitan Small script as an indic script.

**Comment by behdad** [comment]
cc @jfkthame 

Yezidi normally goes through the Universal Shaping Engine, which does intra-cluster `ccmp` I think. If you don't want the Universal Shaping Engine just use the `DFLT` script tag for it instead.

**Comment by cheonhyeongsim** [comment]
Yeah `DFLT` works well (also for Khitan Small Script). But since the very similar Hebrew script can do cross-cluster ligatures by `ccmp`, will Yezidi be also supported in the future version (of USE or of Harfbuzz)?

**Comment by behdad** [comment]
> Yeah `DFLT` works well (also for Khitan Small Script). But since the very similar Hebrew script can do cross-cluster ligatures by `ccmp`, will Yezidi be also supported in the future version (of USE or of Harfbuzz)?

Yezidi is explicitly listed under the USE shaper:

  https://learn.microsoft.com/en-us/typography/script-development/use#writing-system-and-language-tags

If indeed it's like Hebrew and does not adhere to the USE model, it should be reported to Microsoft. In the mean time I suggest using DFLT.

**Comment by dscorbett** [comment]
This is working as intended. The USE model is not just for Indic scripts: it’s supposed to be adequate for any kind of script. You can switch to the default shaper, or use a “standard typographic presentation” feature like 'rclt' or 'rlig'.

**Comment by cheonhyeongsim** [comment]
Thank you for your explanation. Now I see that some of the scripts should use the USE model for shaping. But I have another question that, why does `ccmp` only do intra-cluster substitutions? I did not see anything about this limitation at https://learn.microsoft.com/en-us/typography/opentype/spec/features_ae#tag-ccmp.

**Comment by behdad** [comment]
See https://learn.microsoft.com/en-us/typography/script-development/use#default-glyph-pre-processing-group

**Comment by devosb** [comment]
> Yeah `DFLT` works well (also for Khitan Small Script). But since the very similar Hebrew script can do cross-cluster ligatures by `ccmp`, will Yezidi be also supported in the future version (of USE or of Harfbuzz)?

I am curious what application (other than Crowbar and hb-shape) you are using DFLT with. I have used `DFLT` with a Limbu script font to [avoid a bug](https://github.com/notofonts/limbu/issues/3) (now fixed) in the USE. However, with Notepad and Word on Windows (so DirectWrite was the shaper) the Limbu characters were sent to the USE even though the font did not specify the `limb` shaper. Likewise, DirectWrite sends [Grantha characters to the USE](https://github.com/MicrosoftDocs/typography-issues/issues/738#issuecomment-1177978975
), even though the font specified `tml2` shaper.

**Comment by behdad** [comment]
> I am curious what application (other than Crowbar and hb-shape) you are using DFLT with. I have used `DFLT` with a Limbu script font to [avoid a bug](https://github.com/notofonts/limbu/issues/3) (now fixed) in the USE. However, with Notepad and Word on Windows (so DirectWrite was the shaper) the Limbu characters were sent to the USE even though the font did not specify the `limb` shaper. Likewise, DirectWrite sends [Grantha characters to the USE](https://github.com/MicrosoftDocs/typography-issues/issues/738#issuecomment-1177978975), even though the font specified `tml2` shaper.

That looks like a DWrite bug to me.

**Comment by cheonhyeongsim** [comment]
> I am curious what application (other than Crowbar and hb-shape) you are using DFLT with.

Google Chrome. Just as I mentioned in #4661 and #4662.

> That looks like a DWrite bug to me.

Agree.
